### PR TITLE
Media API

### PR DESCRIPTION
**Pull request by tommorris**
Adds documentation for the Media API.

### CODE REVIEW

**Comment by adambutler** [comment]
🤔  This seems incomplete to me

- [ ] No link on the landing page?
- [ ] No link on the guide anywhere
- [ ] No link on the API page?
- [ ] Broken side navigation

nexmo_developer___documentation

**Comment by tommorris** [comment]
@adambutler I've removed the guide and fixed links to the API doc. The guide can be released later once it's finished.

File: _api/media.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
+---
+title: API reference
+description: Reference guide for the Media API
+api: Media API
+---
+
+# API Reference 
+
+The Media API allows you to manage media files associated with your account and its applications. Authorisation is done using a [JSON Web Token](/concepts/guides/authentication#json-web-tokens-jwt) (JWT) in the `Authorization` header or in a `jwt-token` URL parameter.
```
Comment (adambutler) [inline _api/media.md]: This API supports API Key & Secret also right?

File: _api/media.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+[POST] `https://api.nexmo.com/v3/media`
+
+This endpoint is used to upload media files (usually audio) to the media service. Upon successful upload, you will get back a response with status 201 giving you a URL pointing to the metadata for the media file.
+
+The media file will be associated with both your account and the Application ID of the JWT you use. This enables you to separate media files out by application.
+
+### Request
+
+Requests to this endpoint should be `multipart/form-data`.
```
Comment (adambutler) [inline _api/media.md]: Are we sure this endpoint does not accept `application/json`?

File: _api/media.md [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+|Name|Type|Description|Required|
+|---|---|---|---|
+|`filename`| string | An optional name for the file. | ❎
+|`info`| string (usually JSON object) | Any metadata you wish to associate with the file. | ❎
+|`url`| string (URL) | This can be provided as an alternative to the file data. We will download the file from this URL. This will be stored in `source_url`. | ❎
+
+
+### Response
+
+If the media item has uploaded without error, you will get a `201` status code. The `Location` header will provide a URL for your file.
```
Comment (adambutler) [inline _api/media.md]: Change `201` to `201 Created` (as per write-the-docs guide)

File: _api/media.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+
+|HTTP Code|Description|
+|---|---|
+|201|Media item created|
+|401|Authentication failure|
+|403|Authorisation denied|
+|404|No such item|
+|500|Internal server error|
+
+The URL returned on `201` will be suffixed with `/info` like this:
```
Comment (adambutler) [inline _api/media.md]: Change `201` to `201 Created` (as per write-the-docs guide)

File: _api/media.md [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+|---|---|
+|200|Successful operation|
+|401|Authentication failure|
+|403|Authorisation denied|
+|404|No such item.|
+|500|Internal server error|
+
+A `404` may be returned if no results meet your selection criteria.
+
+
```
Comment (adambutler) [inline _api/media.md]: Remove whitespace

File: _api/media.md [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+## Download media file
+
+[GET] `/v3/media/{media_id}`
```
Comment (adambutler) [inline _api/media.md]: Change from `/v3/media/{media_id}` to `/v3/media/:media_id`

File: _api/media.md [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+
+If the `public_item` property in the file's properties is set to `true`, downloading from this endpoint requires no authentication. If it is set to `false`, you will need to authenticate your request using your application's JSON Web Token in the `Authorization` header or the `jwt-token` parameter.
+
+The number of times a file can be downloaded is determined by the `max_downloads_allowed` property of the file's metadata. If the total number of downloads exceeds that number, you will not be able to download the file unauthenticated. If our request is authenticated with a JSON Web Token, you will be allowed to download the file even if doing so exceeds the `max_downloads_allowed` limit.
+
+### Parameters
+
+|Name|Description|Required|
+|----|-----------|--------|
+|`jwt-token`|A JSON Web Token. Can be used as an alternative to sending an `Authorization` header.|No|
```
Comment (adambutler) [inline _api/media.md]: `No` should use ❎ instead for consistency

File: _api/media.md [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+|304|Not modified|If the media item matches the `If-Modified-Since` or `If-None-Match` header (see above), this is returned.|
+|401|Authentication failure|No Content|
+|403|Authorisation denied|No Content|
+|404|No such item|No Content|
+|500|Internal server error|No Content|
+
+
+## Delete media file
+
+[DELETE] `https://api.nexmo.com/v3/media/{media_id}`
```
Comment (adambutler) [inline _api/media.md]: Change `https://api.nexmo.com/v3/media/{media_id}` to https://api.nexmo.com/v3/media/:media_id`

File: _api/media.md [hunk 9]
```diff
@@ -0,0 +1,231 @@
...
+|204|Media item deleted|No Content|
+|401|Authentication failure|No Content|
+|403|Authorisation denied|No Content|
+|404|No such item|No Content|
+|500|Internal server error|No Content|
+
+
+## Get media file metadata
+
+[GET] `https://api.nexmo.com/v3/media/{media_id}/info`
```
Comment (adambutler) [inline _api/media.md]: Change `https://api.nexmo.com/v3/media/{media_id}/info` to 
`https://api.nexmo.com/v3/media/:media_id/info`

File: _api/media.md [hunk 10]
```diff
@@ -0,0 +1,231 @@
...
+  "media_size": 1024,
+  "time_created": 1502818392405,
+  "time_last_updated": 1502818392405,
+  "public_item": false
+}
+````
+
+## Update media file metadata
+
+[PUT] `https://api.nexmo.com/v3/media/{media_id}/info`
```
Comment (adambutler) [inline _api/media.md]: Change `https://api.nexmo.com/v3/media/{media_id}/info` to `https://api.nexmo.com/v3/media/:media_id/info`

File: _api/media.md [hunk 11]
```diff
@@ -0,0 +1,231 @@
...
+
+|HTTP Code|Description|
+|---|---|
+|200|Successful operation|
+|401|Authentication failure|
+|403|Authorisation denied|
+|404|No such item.|
+|500|Internal server error|
+
+A `404` may be returned if no results meet your selection criteria.
```
Comment (adambutler) [inline _api/media.md]: Change `404` to `404 Not Found`

File: _api/media.md [hunk 12]
```diff
@@ -0,0 +1,231 @@
...
+### Request
+
+Requests to this endpoint should be `multipart/form-data`.
+
+The following parameters can be passed in as form data in addition to the file.
+
+|Name|Type|Description|Required|
+|---|---|---|---|
+|`filename`| string | An optional name for the file. | ❎
+|`info`| string (usually JSON object) | Any metadata you wish to associate with the file. | ❎
```
Comment (adambutler) [inline _api/media.md]: > Usually JSON object

This is vague. I'd expect to see either consistency or further explanation here.

File: _api/media.md [hunk 13]
```diff
@@ -0,0 +1,231 @@
...
+
+# API Reference 
+
+The Media API allows you to manage media files associated with your account and its applications. Authorisation is done using a [JSON Web Token](/concepts/guides/authentication#json-web-tokens-jwt) (JWT) in the `Authorization` header or in a `jwt-token` URL parameter.
+
+## Upload media
+
+[POST] `https://api.nexmo.com/v3/media`
+
+This endpoint is used to upload media files (usually audio) to the media service. Upon successful upload, you will get back a response with status 201 giving you a URL pointing to the metadata for the media file.
```
Comment (adambutler) [inline _api/media.md]: Change 201 to `201 Created`

File: _api/media.md [hunk 14]
```diff
@@ -0,0 +1,231 @@
+---
+title: API reference
```
Comment (adambutler) [inline _api/media.md]: Change title to `Media API Reference`

File: _api/media.md (hunk 1 above)
Comment (tommorris) [reply _api/media.md]: We aren't documenting that as it's not going to be supported.

File: _api/media.md (hunk 2 above)
Comment (tommorris) [reply _api/media.md]: Just tested it. Absolutely guaranteed it doesn't work.

File: _api/media.md (hunk 14 above)
Comment (tommorris) [reply _api/media.md]: Done.

File: _api/media.md (hunk 13 above)
Comment (tommorris) [reply _api/media.md]: Done.

File: _api/media.md (hunk 3 above)
Comment (tommorris) [reply _api/media.md]: Done.

File: _api/media.md (hunk 4 above)
Comment (tommorris) [reply _api/media.md]: Done.

File: _api/media.md (hunk 11 above)
Comment (tommorris) [reply _api/media.md]: Done.

File: _api/media.md (hunk 5 above)
Comment (tommorris) [reply _api/media.md]: Done.

File: _api/media.md (hunk 7 above)
Comment (tommorris) [reply _api/media.md]: Done.

File: _api/media.md (hunk 6 above)
Comment (tommorris) [reply _api/media.md]: Done.

File: _api/media.md (hunk 8 above)
Comment (tommorris) [reply _api/media.md]: Done.

File: _api/media.md (hunk 10 above)
Comment (tommorris) [reply _api/media.md]: Done.

File: _api/media.md (hunk 9 above)
Comment (tommorris) [reply _api/media.md]: Done.

File: _api/media.md (hunk 12 above)
Comment (tommorris) [reply _api/media.md]: Okay, had a crack at fixing this.

**Comment by tommorris** [comment]
@adambutler Fixed all the things. Want to take another look?

**Pull request closed without merging**